utils/onnxmlirrun.py

```diff
@@ -0,0 +1,186 @@
+﻿#!/usr/bin/python
+# Copyright 2019-2023 The IBM Research Authors.
+
+import os
+import sys
+import onnx
+import time
+import signal
+import subprocess
+import numpy as np
+import tempfile
+
+from onnx import numpy_helper
+from onnx.mapping import TENSOR_TYPE_TO_NP_TYPE
+from collections import OrderedDict
+
+# This file provide utility to compile and run onnx model with onnx-mlir,
+# in an existing python env, such as Pytorch, tensorflow or SciKit learn.
+# The interface is delibrately designed similar to onnxruntime to reduce user's
+# burden of learning and code change.
+# Lots of code is inherited from utils/RunONNXModel.py, which is a 
+# "standalone" python script to use onnx-mlir compiler.
+# In future, this file will evolved to be part of onnx-mlir python package. 
+# An example:
+
+"""
+import onnxmlirrun
+import numpy as np
+
+a = np.random.rand(3, 4, 5).astype(np.float32)
+b = np.random.rand(3, 4, 5).astype(np.float32)
+session = onnxmlirrun.InferenceSession("test_add.onnx")
+outputs = session.run(None, {"a": a, "b":b})
+print(outputs)
+"""
+
+
+
+class InferenceSession:
+
+    def __init__(self, model_path, target="cpu", **kwarg):
+        self.target = target
+        if "options" in kwarg :
+          self.options = kwarg["options"]
+        else :
+          self.options = ""
+
+        print("target : ", self.target)
+        print(" options : ", self.options)
+
+        # Initialize status
+
+        self.compiled = False
+        self.loaded = False
+
+        # Initialize parameters
+
+        self.VERBOSE = os.environ.get('VERBOSE', False)
+        self.input_model_path = model_path
+
+        # name for the compiled library in temporary directory
+
+        self.temp_lib_name = 'model'
+        if not os.environ.get('ONNX_MLIR_HOME', None):
```

Comment:
If its not easy for a python script to set environment variable, would it be a good idea to pass it as an optional parameter, like you did for options?

Reply:
added an optional  parameter for the compiler path. We may be able to include the compiler in the python package in future.